middleware/templates/templates_test.go

```diff
@@ -95,6 +96,30 @@ func Test(t *testing.T) {
 	}
 
 	/*
+	* Test tmpl on /images/img2.htm
+	 */
+	req, err = http.NewRequest("GET", "/images/img2.htm", nil)
+	if err != nil {
+		t.Fatalf("Could not create HTTP request: %v", err)
+	}
+
+	rec = httptest.NewRecorder()
+
+	tmpl.ServeHTTP(rec, req)
+
+	if rec.Code != http.StatusOK {
+		t.Fatalf("Test: Wrong response code: %d, should be %d", rec.Code, http.StatusOK)
+	}
+
+	respBody = rec.Body.String()
+	expectedBody = `<!DOCTYPE html><html><head><title>img</title></head><body>{{.Include "header.html"}}</body></html>
```

Comment:
This is clever -- so the test asserts that the template processing did _not_ change the body.

What about adding a test that does change the body using the custom delimiter (for good measure)? Hmm, maybe you could even test this without having to put a new file on the file system -- just pass a []byte or some reader as input.